### PR TITLE
feat: (IAC-801): added AWS EBS and EFS encryption

### PR DESCRIPTION
**Pull request by anthoday**
<h2>Feature</h2>

Added the option for customers to enable encryption for AWS Elastic Block Store (EBS) volumes and Elastic File System (EFS) from within the Terraform module. With this option, customers can control the encryption of their data directly from within the Terraform module, eliminating the need to go to the AWS management console and set up encryption manually.

<h2>Changes</h2>
The following elements were added to allow customers to enable EBS and EFS encryption:

* In the sub-directory './modules/aws_vm/variables.tf'
   * Created a variable for enabling EBS encryption - set to false by default.
*  In the sub-directory './modules/aws_vm/main.tf'
   * Set the 'encrypted' flag within 'aws_instance' and 'aws_ebs_volume' resources to 'var.enable_ebs_encryption'.
* In the root directory - 'variables.tf'
   * Created variables for enabling EBS and EFS encryption - set to 'false' by default.
*  In the root directory - 'locals.tf'
   * Set the 'encrypted' flag within the 'default_node_pool' and 'user_node_pool' local resources to 'var.enable_ebs_encryption'.
* In the root directory - 'vms.tf'
   * Set the 'encrypted' flag within the 'aws_efs_file_system' to 'var.enable_efs_encryption'.
   * Set the 'enable_ebs_encryption' flag within the 'Jump BOX' and 'NFS Server VM' modules to 'var.enable_ebs_encryption'.
* In the sub-directory - './examples'
   * Within all example tfvars files - added the option to enable EBS and EFS encryption.

<h2>Test</h2>
See internal ticket for details and test artifacts:

1. EBS encryption and instance-based NFS (EBS encrypted)
   * EBS encryption done by setting "enable_ebs_encryption = true" - within local terraform.tfvars file.
   * Successfully encrypted AWS infrastructure and deployed Viya4.

2. EBS encryption and EFS encryption
   * EBS encryption done by setting "enable_ebs_encryption = true" - within local terraform.tfvars file.
   * EFS encryption done by setting "enable_efs_encryption = true" - within local terraform.tfvars file.
   * Successfully encrypted AWS infrastructure and deployed Viya4.

### CODE REVIEW

File: examples/sample-input-byo.tfvars [hunk 1]
```diff
@@ -45,6 +45,10 @@ default_nodepool_custom_data            = ""
 ## General
 efs_performance_mode                    = "maxIO"
 storage_type                            = "standard"
+# To enable EBS encyption - set to 'true'. Variable is set to 'false' by default.
+enable_ebs_encryption = false
+# To enable EFS encyption - set to 'true'. Variable is set to 'false' by default.
+enable_efs_encryption = false
```
Comment (thpang) [inline examples/sample-input-byo.tfvars:51]: Just wondering why we included these elements in the sample files if the values are only applying the default value of `false`? Does the CONFIG-VARS.md file not cover the addition of these flags? This comment holds for all sample files where the changes were made.

File: locals.tf [hunk 2]
```diff
@@ -42,6 +42,7 @@ locals {
             volume_type                     = var.default_nodepool_os_disk_type
             volume_size                     = var.default_nodepool_os_disk_size
             iops                            = var.default_nodepool_os_disk_iops
+            encrypted   = var.enable_ebs_encryption
```
Comment (thpang) [inline locals.tf:45]: New element needs to be aligned with the old code or `terraform fmt` needs to be run to clean up the tf files in this repo.

File: locals.tf [hunk 3]
```diff
@@ -86,6 +87,7 @@ locals {
               volume_type                   = np_value.os_disk_type
               volume_size                   = np_value.os_disk_size
               iops                          = np_value.os_disk_iops
+              encrypted   = var.enable_ebs_encryption
```
Comment (thpang) [inline locals.tf:90]: Formatting issue as noted above.

File: modules/aws_vm/main.tf [hunk 4]
```diff
@@ -104,4 +105,5 @@ resource "aws_ebs_volume" "raid_disk" {
   type              = var.data_disk_type
   iops              = var.data_disk_iops
   tags              = merge(var.tags, tomap({ Name: "${var.name}-vm" }))
+  encrypted             = var.enable_ebs_encryption
```
Comment (thpang) [inline modules/aws_vm/main.tf:108]: formatting

File: modules/aws_vm/variables.tf [hunk 5]
```diff
@@ -78,3 +78,7 @@ variable "os_disk_iops" {
 variable "subnet_id" {
   type = string
 }
+
+variable "enable_ebs_encryption" {
+  default = false
+}
```
Comment (thpang) [inline modules/aws_vm/variables.tf:84]: Provide a description here as to the purpose of the variable.

File: vms.tf [hunk 6]
```diff
@@ -15,6 +15,7 @@ resource "aws_efs_file_system" "efs-fs" {
   creation_token   = "${var.prefix}-efs"
   performance_mode = var.efs_performance_mode
   tags             = merge(var.tags, { "Name": "${var.prefix}-efs" })
+  encrypted = var.enable_efs_encryption
```
Comment (thpang) [inline vms.tf:18]: formatting

**Review by thpang — CHANGES_REQUESTED** [review]

File: examples/sample-input-byo.tfvars (hunk 1 above)
Comment (anthoday) [reply examples/sample-input-byo.tfvars:51]: CONFIG-VARS.md does not cover these flags. Should I remove them from the tfvars files and cover them in the CONFIG-VARS.md?

File: vms.tf [hunk 7]
```diff
@@ -82,6 +83,7 @@ module "jump" {
   vm_type        = var.jump_vm_type
   vm_admin       = var.jump_vm_admin
   ssh_public_key = local.ssh_public_key
+  enable_ebs_encryption = var.enable_ebs_encryption
```
Comment (dhoucgitter) [inline vms.tf:86]: formatting

File: vms.tf [hunk 8]
```diff
@@ -138,6 +140,7 @@ module "nfs" {
   vm_type        = var.nfs_vm_type
   vm_admin       = var.nfs_vm_admin
   ssh_public_key = local.ssh_public_key
+  enable_ebs_encryption = var.enable_ebs_encryption
```
Comment (dhoucgitter) [inline vms.tf:143]: formatting

File: locals.tf (hunk 2 above)
Comment (anthoday) [reply locals.tf:45]: Acknowledged. Formatting corrected.

File: locals.tf (hunk 3 above)
Comment (anthoday) [reply locals.tf]: Acknowledged. Formatting corrected.

File: modules/aws_vm/main.tf (hunk 4 above)
Comment (anthoday) [reply modules/aws_vm/main.tf:108]: Acknowledged. Formatting corrected.

File: vms.tf (hunk 6 above)
Comment (anthoday) [reply vms.tf]: Acknowledged. Formatting corrected.

File: vms.tf (hunk 8 above)
Comment (anthoday) [reply vms.tf:143]: Acknowledged. Formatting corrected.

File: vms.tf (hunk 7 above)
Comment (anthoday) [reply vms.tf:86]: Acknowledged. Formatting corrected.

File: modules/aws_vm/variables.tf (hunk 5 above)
Comment (anthoday) [reply modules/aws_vm/variables.tf]: Acknowledged. Description added.

File: examples/sample-input-byo.tfvars (hunk 1 above)
Comment (anthoday) [reply examples/sample-input-byo.tfvars]: Added EBS and EFS elements to CONFIG-VARS.md. Updates are up now.

File: docs/CONFIG-VARS.md [hunk 9]
```diff
@@ -274,6 +274,14 @@ When `storage_type=ha`, the [AWS Elastic File System](https://aws.amazon.com/efs
 | <div style="width:50px">Name</div> | <div style="width:150px">Description</div> | <div style="width:50px">Type</div> | <div style="width:75px">Default</div> | <div style="width:150px">Notes</div> |
 | :--- | :--- | :--- | :--- | :--- |
 | efs_performance_mode | EFS performance mode | string | generalPurpose | Supported values are `generalPurpose` or `maxIO` |
+| enable_efs_encryption | Enable encryption on EFS file systems | bool | false | |
+
+## AWS Elastic Block Store (EBS)
+
```
Comment (thpang) [inline docs/CONFIG-VARS.md:280]: Indent to level 3 with ### and add a description.

**Review by thpang — CHANGES_REQUESTED** [review]

File: docs/CONFIG-VARS.md (hunk 9 above)
Comment (anthoday) [reply docs/CONFIG-VARS.md]: Acknowledged. CONFIG-VARS.md updated.

**Review by thpang — APPROVED** [review]

**Review by riragh — APPROVED** [review]

**Pull request merged**